app/src/main/java/org/sil/hearthis/MainActivity.java

```diff
@@ -1,76 +1,220 @@
 package org.sil.hearthis;
 
-import org.apmem.tools.layouts.FlowLayout.LayoutParams;
-import org.sil.hearthis.R;
+import org.json.JSONObject;
 import org.sil.palaso.Graphite;
 
 import Script.BookInfo;
+import Script.IScriptProvider;
 import Script.Project;
 import Script.RealScriptProvider;
 import Script.SampleScriptProvider;
 import android.app.Activity;
 import android.content.Context;
-import android.content.DialogInterface;
-import android.content.DialogInterface.OnClickListener;
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.os.Bundle;
-import android.renderscript.Sampler;
+import android.support.v7.app.ActionBarActivity;
+import android.view.Gravity;
 import android.view.LayoutInflater;
 import android.view.Menu;
+import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.GridView;
+import android.widget.TextView;
+import android.widget.Toast;
 
-public class MainActivity extends Activity {
+import java.util.ArrayList;
+
+public class MainActivity extends ActionBarActivity
+{
+	Menu mainMenu;
 
 	@Override
-	protected void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState);
+	protected void onCreate(Bundle state)
+	{
+		super.onCreate(state);
+
         Graphite.loadGraphite();
 		setContentView(R.layout.activity_main);
+
 		// Todo: scan org.sil.hearthis/files for folders containing info.txt and open first
-		// Todo: remember last project
-		// Todo: if no real project available use SampleScriptProvider.
-		RealScriptProvider scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+
+		IScriptProvider scripture;
+		try
+		{
+			scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+		}
+		catch (Exception ex)
```

Comment:
In general, I consider it bad practice to catch all exceptions like this. For example, if I introduce some coding bug into the RealScriptProvider constructor and generate a null reference exception, I don't want it ignored. Could you catch more specific exceptions that really make this response appropriate, like FileNotFound, or maybe IOException?